security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -309,7 +309,11 @@ func (sc *SecretController) saAdded(obj interface{}) {
 
 // Handles the event where a service account is deleted.
 func (sc *SecretController) saDeleted(obj interface{}) {
-	acct := obj.(*v1.ServiceAccount)
+	acct, ok := obj.(*v1.ServiceAccount)
+	if !ok {
+		k8sControllerLog.Warnf("Failed to convert to serviceaccount object: %v", obj)
```

Comment:
With the client library link you provided, I now understand, if correctly,

- `onDelete` can be fed with `DeletedFinalStateUnknown` type object, in case of **missing deletion**
- Citadel thus should not proceed in that case.
- that's why we create DeletedFinalStateUnknown obj in the test code and call it `missing deletion`.

Could you add a bit more comments above this line? Thus we can also attribute why unit test needs a `DeletedFinalStateUnknown` obj, and using `missing..` wording. (they dont' appear in the impl code, so hard to tie their relationship right away.)

Reply:
added comments